crates/net/rpc-types/src/admin.rs

```diff
@@ -0,0 +1,83 @@
+use reth_primitives::{NodeRecord, PeerId, H256, U256};
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::BTreeMap,
+    net::{IpAddr, SocketAddr},
+};
+
+/// Represents the `admin_nodeInfo` response, which can be queried for all the information
+/// known about the running node at the networking granularity.
+///
+/// Note: this format is not standardized.  Reth follows geth's format,
+/// see: https://geth.ethereum.org/docs/interacting-with-geth/rpc/ns-admin
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NodeInfo {
+    /// Enode in URL format.
+    pub enode: NodeRecord,
+    /// ID of the local node.
+    pub id: PeerId,
+    /// IP of the local node.
+    pub ip: IpAddr,
+    /// Address exposed for listening for the local node.
+    #[serde(rename = "listenAddr")]
+    pub listen_addr: SocketAddr,
+    /// Local node client name.
+    pub name: String,
+    /// Ports exposed by the node for discovery and listening.
+    pub ports: Ports,
+    /// Networking protocols being run by the local node.
+    pub protocols: BTreeMap<String, ProtocolInfo>,
+}
+
+impl NodeInfo {
+    /// Creates a new instance of `NodeInfo`.
+    pub fn new(enr: NodeRecord) -> NodeInfo {
+        let mut protocol_info = BTreeMap::from([(
+            "eth".into(),
+            ProtocolInfo::Eth(EthProtocolInfo {
+                difficulty: todo!(),
+                head: todo!(),
+                network: 0,
+                genesis: todo!(),
```

Comment:
Can get that from the status message of the network component? I think we had a getter? @mattsse

Reply:
yeh will open an issue to add a function for getting the status